packages/stack-ui/src/components/brand-icons.tsx

```diff
@@ -0,0 +1,215 @@
+      </g>
+    </svg>
+  );
+}
+
+export function Apple({ iconSize } : { iconSize: number} ) {
+  return (
+    <svg fill="#fff" height={iconSize} width={iconSize} version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 22.773 22.773">
+      <path d="M15.769,0c0.053,0,0.106,0,0.162,0c0.13,1.606-0.483,2.806-1.228,3.675c-0.731,0.863-1.732,1.7-3.351,1.573c-0.108-1.583,0.506-2.694,1.25-3.561C13.292,0.879,14.557,0.16,15.769,0z"/>
+      <path d="M20.67,16.716c0,0.016,0,0.03,0,0.045c-0.455,1.378-1.104,2.559-1.896,3.655c-0.723,0.995-1.609,2.334-3.191,2.334c-1.367,0-2.275-0.879-3.676-0.903c-1.482-0.024-2.297,0.735-3.652,0.926c-0.155,0-0.31,0-0.462,0c-0.995-0.144-1.798-0.932-2.383-1.642c-1.725-2.098-3.058-4.808-3.306-8.276c0-0.34,0-0.679,0-1.019c0.105-2.482,1.311-4.5,2.914-5.478c0.846-0.52,2.009-0.963,3.304-0.765c0.555,0.086,1.122,0.276,1.619,0.464c0.471,0.181,1.06,0.502,1.618,0.485c0.378-0.011,0.754-0.208,1.135-0.347c1.116-0.403,2.21-0.865,3.652-0.648c1.733,0.262,2.963,1.032,3.723,2.22c-1.466,0.933-2.625,2.339-2.427,4.74C17.818,14.688,19.086,15.964,20.67,16.716z"/>
+    </svg>
+  );
+}
+
+export function X({ iconSize } : { iconSize: number} ) {
+  return (
+    <svg aria-label="X" viewBox="0 0 1200 1227" width={iconSize} height={iconSize} xmlns="http://www.w3.org/2000/svg">
+      <path fill="#FFFFFF" d="M714.163 519.284L1160.89 0H1055.03L667.137 450.887L357.328 0H0L468.492 681.821L0 1226.37H105.866L515.491 750.218L842.672 1226.37H1200L714.137 519.284H714.163ZM569.165 687.828L521.697 619.934L144.011 79.6944H306.615L611.412 515.685L658.88 583.579L1055.08 1150.3H892.476L569.165 687.854V687.828Z"/>
+    </svg>
+  );
+}
+
+export function Mapping({
+  provider,
+  iconSize,
+}: {
+  provider: string,
+  iconSize: number,
+}) {
+  switch (provider) {
+    case "google": {
+      return <Google iconSize={iconSize} />;
+    }
+    case "github": {
+      return <GitHub iconSize={iconSize} />;
+    }
+    case "facebook": {
+      return <Facebook iconSize={iconSize} />;
+    }
+    case "microsoft": {
+      return <Microsoft iconSize={iconSize} />;
+    }
+    case "spotify": {
+      return <Spotify iconSize={iconSize} />;
+    }
+    case "discord": {
+      return <Discord iconSize={iconSize} />;
+    }
+    case "gitlab": {
+      return <Gitlab iconSize={iconSize} />;
+    }
+    case "bitbucket": {
+      return <Bitbucket iconSize={iconSize} />;
+    }
+    case "linkedin": {
+      return <LinkedIn iconSize={iconSize} />;
+    }
+    case "apple": {
+      return <Apple iconSize={iconSize} />;
+    }
+    case "x": {
+      return <X iconSize={iconSize} />;
+    }
+    default: {
+      return <></>;
```

Comment:
Whenever you can, throw errors instead of showing empty or loading values

```suggestion
      return throw new StackAssertionError(`Icon not found for provider: ${provider}`);
```

Reply:
Done